vault/identity_store_oidc_provider.go

```diff
@@ -327,3 +393,129 @@ func (i *IdentityStore) pathOIDCScopeExistenceCheck(ctx context.Context, req *lo
 
 	return entry != nil, nil
 }
+
+// pathOIDCCreateUpdateClient is used to create a new named client or update an existing one
+func (i *IdentityStore) pathOIDCCreateUpdateClient(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	ns, err := namespace.FromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	name := d.Get("name").(string)
+
+	var client namedClient
+	if req.Operation == logical.UpdateOperation {
+		entry, err := req.Storage.Get(ctx, namedClientPath+name)
+		if err != nil {
+			return nil, err
+		}
+		if entry != nil {
+			if err := entry.DecodeJSON(&client); err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	if redirectURIsRaw, ok := d.GetOk("redirect_uris"); ok {
+		client.RedirectURIs = redirectURIsRaw.([]string)
+	} else if req.Operation == logical.CreateOperation {
+		client.RedirectURIs = d.Get("redirect_uris").([]string)
+	}
+
+	if assignmentsRaw, ok := d.GetOk("assignments"); ok {
+		client.Assignments = assignmentsRaw.([]string)
+	} else if req.Operation == logical.CreateOperation {
+		client.Assignments = d.Get("assignments").([]string)
+	}
+
+	if keyRaw, ok := d.GetOk("key"); ok {
```

Comment:
Similar to identity token roles, we should consider if we want to validate that the referenced key exists. I _think_ that's the behavior that we want, especially if that's what we plan to do for identity token roles.

Reply:
If that's the path we take, then we'll likely need to add another check on key deletion (similar to [identity_store_oidc.go#L635-L645](https://github.com/hashicorp/vault/blob/main/vault/identity_store_oidc.go#L635-L645)) to ensure that the key isn't being used by existing clients. This can be another dev sync topic.

Reply:
done